india_compliance/gst_india/setup/__init__.py

```diff
@@ -225,3 +226,14 @@ def _get_custom_fields_map(*custom_fields_list):
             result.setdefault(doctypes, []).extend(fields)
 
     return result
+
+
+def remove_variant_hsn_map():
+    row = frappe.db.exists("Variant Field", {"field_name": "gst_hsn_code"})
+    if row:
+        ivs = frappe.get_doc("Item Variant Settings")
+        to_remove = [d for d in ivs.fields if d.field_name == "gst_hsn_code"]
+        ivs.fields.remove(to_remove[0])
+        for index, item in enumerate(ivs.fields):
+            item.idx = index
+        ivs.save()
```

Comment:
```suggestion
       frappe.delete_doc("Variant Field", row)
```

Reply:
I have changed the code to the suggested one, although it can mess up the idx on the table:
![image](https://user-images.githubusercontent.com/52111700/213145195-6e09656b-29c0-4bd0-a693-f1e7a1d2e724.png)